wlauto/workloads/skypeecho/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,207 @@
+        String useridResoureId = PACKAGE_ID + "sign_in_userid";
+        String nextButtonResourceId = PACKAGE_ID + "sign_in_next_btn";
+        UiObject useridField = new UiObject(new UiSelector().resourceId(useridResoureId));
+        UiObject nextButton = new UiObject(new UiSelector().resourceId(nextButtonResourceId));
+        useridField.setText(username);
+        nextButton.clickAndWaitForNewWindow();
+
+        String skypenameResoureId = PACKAGE_ID + "signin_skypename";
+        String passwordResoureId = PACKAGE_ID + "signin_password";
+        String signinButtonResourceId = PACKAGE_ID + "sign_in_btn";
+        // UiObject skypenameField = new UiObject(new UiSelector().resourceId(skypenameResoureId));
+        UiObject passwordField = new UiObject(new UiSelector().resourceId(passwordResoureId));
+        UiObject signinButton = new UiObject(new UiSelector().resourceId(signinButtonResourceId));
+        // skypenameField.setText(username);
+        passwordField.setText(password);
+        signinButton.clickAndWaitForNewWindow();
+    }
+
+    private void selectContact(String name, String id) throws Exception {
+        Timer timer = new Timer();
+        timer.start();
+        UiObject peopleTab;
+        // Open the 'People' tab aka contacts view
+        // On phones, it is represented by an image with description
+        // On tablets, it the full text is shown without a description
+        try {
+            peopleTab = getUiObjectByDescription("People", TEXT_VIEW);
+        } catch (UiObjectNotFoundException e) {
+            peopleTab = getUiObjectByText("People", TEXT_VIEW);
+        }
+        peopleTab.click();
+
+        // On first startup, the app may take a while to load the display name,
+        // so try twice before declaring failure
+        UiObject contactCard;
+        try {
+            contactCard = getUiObjectByText(name, TEXT_VIEW);
+        } catch (UiObjectNotFoundException e) {
+            contactCard = getUiObjectByText(name, TEXT_VIEW);
+            // contactCard = getUiObjectByText(id, TEXT_VIEW);
+        }
+        contactCard.clickAndWaitForNewWindow();
+        timer.end();
+        results.put("select_contact", timer);
+    }
+
+    private void voiceCallTest(int duration) throws Exception {
+        String testTag = "voice_call";
+        Timer timer = new Timer();
+        timer.start();
+        makeCall(duration, false, testTag);
+        timer.end();
+        results.put(testTag, timer);
+    }
+
+    private void videoCallTest(int duration) throws Exception {
+        String testTag = "video_call";
+        Timer timer = new Timer();
+        timer.start();
+        makeCall(duration, true, testTag);
+        timer.end();
+        results.put(testTag, timer);
+    }
+
+    private void makeCall(int duration, boolean video, String testTag) throws Exception {
+        String viewName = "com.skype.android.app.calling.CallActivity";
```

Comment:
We made a small change to the log file names we post process.  To bring this into line could you use...
String runName = String.format(TestTag + "_" + pair.getKey());
            String gfxInfologName =  String.format(runName + "_gfxInfo.log");
            String surfFlingerlogName =  String.format(runName + "_surfFlinger.log");